### PR TITLE
Fix bugs in external flash flashing

### PR DESCRIPTION
**Pull request by andyp1per**
Allow the flashing of external flash images to boards with external flash

Tested on an SPRacingH7

### CODE REVIEW

File: ExtLibs/px4uploader/Uploader.cs [hunk 1]
```diff
@@ -500,6 +500,8 @@ public bool __trySync()
             byte c = __recv()[0];
             if (c != (byte)Code.INSYNC)
                 return false;
+
+            c = __recv()[0];
```
Comment (meee1) [inline ExtLibs/px4uploader/Uploader.cs:504]: Not sure about this change, why the extra read?

File: ExtLibs/px4uploader/Uploader.cs [hunk 2]
```diff
@@ -742,25 +794,52 @@ public void __verify_extf(Firmware fw)
...
+
             this.port.ReadTimeout = 1000; // 1 sec
 
-            int expect_crc = fw.crc(self.fw_maxsize);
-            __send(new byte[] {(byte)Code.GET_CRC,
-				(byte) Code.EOC});
-            int report_crc = __recv_int();
-            self.__getSync();
+            if (self.fw_maxsize > 0 && false)
+            {
```
Comment (meee1) [inline ExtLibs/px4uploader/Uploader.cs:807]: && false?

File: ExtLibs/px4uploader/Uploader.cs (hunk 1 above)
Comment (andyp1per) [reply ExtLibs/px4uploader/Uploader.cs:504]: Look at the code for sync() - it is absolutely required. This function clealy never worked.

File: ExtLibs/px4uploader/Uploader.cs (hunk 2 above)
Comment (andyp1per) [reply ExtLibs/px4uploader/Uploader.cs:807]: oops, will remove

File: ExtLibs/px4uploader/Uploader.cs (hunk 1 above)
Comment (meee1) [reply ExtLibs/px4uploader/Uploader.cs:504]: indeed, a 7 year old bug.....

Comment (andyp1per) [reply ExtLibs/px4uploader/Uploader.cs:504]: Hah, yes - I only found it by comparing with the python uploader

File: ExtLibs/px4uploader/Uploader.cs (hunk 2 above)
Comment (andyp1per) [reply ExtLibs/px4uploader/Uploader.cs:807]: Fixed

**Comment by andyp1per** [comment]
Thanks @meee1 !

**Pull request merged**